src/components/countryDetails/CountryDetails.tsx

```diff
@@ -1,49 +1,122 @@
-	const { inputValue, setInputValue } = useCountriesContext();
-	const { commonName } = useParams();
+	const { inputValue, setInputValue, countries, fetchCountries } =
+		useCountriesContext();
 	const countryData = useLoaderData() as CountryDataTypes[];
 
-	const handleClick = (): void => {
+	useEffect(() => {
+		if (!countries?.length) {
+			fetchCountries();
+		}
+	}, []);
+
+	const handleClick = () => {
 		setInputValue(inputValue);
 	};
 
 	return (
-		<div>
-			<Link onClick={handleClick} to="/">
-				Go back
-			</Link>
-			countryDetails
-			<p>{commonName}</p>
-			<div>
-				{countryData?.map((country) => {
-					return <p key={country.name.common}>{country.name.common}</p>;
-				})}
-			</div>
-		</div>
-	);
-};
+		<StyledSection>
+			<StyledDivWithFlag>
+				<StyledLink onClick={handleClick} to="/">
+					<StyledArrowIcon />
+					Back
+				</StyledLink>
 
-export const countryDetailsLoader: LoaderFunction = async ({ params }) => {
-	const { commonName } = params;
-	const res = await fetch(`https://restcountries.com/v3.1/name/${commonName}`);
+				<div>
+					{countryData?.map(({ name, flags }) => {
+						return (
+							<StyledFlagImage
+								src={flags.png}
+								alt={flags.alt}
+								key={name.common}
+							/>
+						);
+					})}
+				</div>
+			</StyledDivWithFlag>
+			<StyledDivWithCountryData>
+				{countryData?.map(
+					({
+						name,
+						population,
+						region,
+						subregion,
+						capital,
+						tld,
+						currencies,
+						languages,
+						altSpellings,
+						borders,
+					}) => {
+						const borderCountriesArr = borders?.map((singleBorder) => {
+							return countries?.find((singleCountry) => {
+								return singleCountry.cca3 === singleBorder;
+							});
+						});
 
-	if (!res.ok) {
-		throw Error("Could not find that country.");
-	}
-
-	return res.json();
+						return (
+							<div key={name.common}>
+								<StyledCountryInfoWrapper>
+									<StyledCountryName>{name.common}</StyledCountryName>
+									<StyledCountryDataWrapper>
+										<LeftColumnWithCountryInfo
+											nativeName={altSpellings}
+											population={population}
+											region={region}
+											subregion={subregion}
+											capital={capital}
+										/>
+										<RightColumnWithCountryInfo
+											tld={tld}
+											currencies={currencies}
+											languages={languages}
+										/>
+									</StyledCountryDataWrapper>
+								</StyledCountryInfoWrapper>
+								<StyledWrapperWithButtons>
+									<StyledCategoryName>Border Countries:</StyledCategoryName>{" "}
+									{borderCountriesArr?.length ? (
+										borderCountriesArr.map((country, index) => {
+											return (
+												<StyledButtons
+													to={`/${country?.name.common}`}
+													key={index}
+													type="button"
+												>
+													{country?.name.common}
+												</StyledButtons>
+											);
+										})
+									) : (
+										<NoBorderCountries />
```

Comment:
Can you avoid this ternary by introducing the early return pattern?

Reply:
changed. Moved to seperate component.